src/features/supervisor-app/hooks/supervisor-app.ts

```diff
@@ -116,14 +115,30 @@ async function checkSupervisorReleaseUpgrades(
 		},
 	});
 
-	for (const release of releases) {
+	let doingUpgrade = false;
+
+	for (const release of oldReleases) {
 		const oldVersion = release.release_version;
-		if (semver.lt(newSupervisorVersion, oldVersion)) {
+		if (semver.gt(newSupervisorVersion, oldVersion)) {
+			doingUpgrade = true;
+		} else if (semver.lt(newSupervisorVersion, oldVersion)) {
 			throw new BadRequestError(
 				`Attempt to downgrade supervisor, which is not allowed`,
 			);
 		}
 	}
+
+	return doingUpgrade;
+}
+
+async function getRelease(api: sbvrUtils.PinejsClient, releaseId: number) {
+	return await api.get({
+		resource: 'release',
+		id: releaseId,
```

Comment:
As I understand it, this query could return any release and the clients calling it just assume it is from the Supervisor app. Can we improve on this?

Reply:
Use of this method will change a bit with the next PR.  If it still seems like we are making unnecessarily wide assumptions, let's address it there?